### ISSUE TITLE
[VPAT][A11y] AutoComplete dropdown option is missing button role

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#173058]
b/429089966

Run the https://github.com/flutter/flutter/tree/master/dev/a11y_assessments and navigate to autocomplete demo

<img width="1041" height="1887" alt="Image" src="https://github.com/user-attachments/assets/4b4b8e6c-6c1c-40cf-9dc7-e9cb00ec7408" />

It should announce 'apple, button'

**Comment by chunhtai** [comment]
should be able to fix by adding a semantics(button: true) around the options

**Comment by srivats22** [comment]
Can I take it? 😀

**Comment by chunhtai** [comment]
feel free to send a pr, I don't think anyone is working on this at the moment

**Comment by srivats22** [comment]
Got a question on this... I am making change only for the autocomplete demo? or am I making the change in the underlying rawautocomplete file?
cause when I try adding this piece of code in the demo
```dart
    optionsViewBuilder: (BuildContext context, AutocompleteOnSelected<String> onSelected, Iterable<String> options) {
                return Semantics(
                  // This is where the accessibility label is set.
                  button: true,
                  child: Align(
                    alignment: Alignment.topLeft,
                    child: Material(
                      elevation: 4.0,
                      child: SizedBox(
                        height: 200.0,
                        child: ListView.builder(
                          padding: EdgeInsets.zero,
                          itemCount: options.length,
                          itemBuilder: (BuildContext context, int index) {
                            final String option = options.elementAt(index);
                            return GestureDetector(
                              onTap: () {
                                onSelected(option);
                              },
                              child: ListTile(
                                title: Text(option),
                              ),
                            );
                          },
                        ),
                      ),
                    ),
                  ),
                );
              },
```

it doesnt work as expected I still only hear apple and not apple button...

**Comment by chunhtai** [comment]
I recommend we fix for all autocomplete, and only fix the a11y app if the expected behavior can't be fixed in the flutter widget.

you can dump the semantics tree by press shift+S in command line to inspect whether the tree looks expected after applying the patch

**Comment by srivats22** [comment]
Understood thank you, so I will go ahead and make the change in the underlying auto complete widget. And I will also use the command you mentioned to see if it's working as expected. I might take some time to deliver it

**Comment by srivats22** [comment]
Hi,
I opened the following pr: https://github.com/flutter/flutter/pull/173297 for this issue

The Flutter golden test appears to be still running... is it common for it to run for this long?
